portfolio/src/main/webapp/script-map.js

```diff
@@ -0,0 +1,144 @@
+// limitations under the License.
+
+let map;
+const flagIcon = 'https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png';
+
+/* Editable marker that displays when a user clicks in the map. */
+let editMarker;
+
+/** Creates a map that shows travel markers and allows user to input markers. */
+function createMap() {
+  map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 39.329858, lng: -76.620540}, zoom: 5});
+  
+  // When the user clicks in the map, show a marker with a text box the user can edit.
+  map.addListener('click', (event) => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+
+  createMarkerForDisplay(
+      39.329858, -76.620540,
+      'I go to school here.', 'Johns Hopkins University', false)
+  createMarkerForDisplay(
+      32.003407, 118.734956,
+      'This is where I was born.', 'Nanjing', false)
+  createMarkerForDisplay(
+      37.423829, -122.092154,
+      'This is where my internship was supposed to be, but now it\'s remote.', 'Google', false);
+  createMarkerForDisplay(
+      23.117487, -82.373375,
+      'I went there with my friends. It was great fun.', 'Havana', false);
+  createMarkerForDisplay(
+      38.642805, -90.195679,
+      'I worked here during summer, 2019.', 'St Louis', false);
+
+  fetchMarkers();
+}
+
+/** 
+ * Adds a marker that shows an info window when clicked. 
+ * @param {number} lat Latitude of the marker position.
+ * @param {number} lng Longitude of the marker position.
+ * @param {string} content The content of the marker, which is displayed onclick.
+ * @param {string=} title Optional title of the marker, which is displayed on hover.
+ * @param {boolean=} defaultIcon Whether to display the default marker icon or the flagIcon. 
+ */
+function createMarkerForDisplay(lat, lng, content, title = '', defaultIcon = true) {
+  let marker;
+  if (defaultIcon == false) {
+    marker = new google.maps.Marker(
+        {position: {lat: lat, lng: lng}, map: map, title: title, icon: flagIcon});
+  } else {
+    marker = new google.maps.Marker(
+      {position: {lat: lat, lng: lng}, map: map});
+  }
+
+  const infoWindow = new google.maps.InfoWindow({content: content});
+  marker.addListener('click', () => {
+    infoWindow.open(map, marker);
+  });
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers() {
+  fetch('/markers').then(response => response.json()).then((markers) => {
+    markers.forEach(
+        (marker) => {
+            createMarkerForDisplay(marker.lat, marker.lng, marker.content)});
+  });
+}
+
+/** 
+ * Creates a marker that shows a textbox the user can edit. 
+ * @param {number} lat Latitude of the marker position.
+ * @param {number} lng Longitude of the marker position.
+ */
+function createMarkerForEdit(lat, lng) {
+  // If we're already showing an editable marker, then remove it.
```

Comment:
You want to say what you're trying to do in words, and sound as little like code/pseudocode as possible :)

```suggestion
  // Remove any editable marker that is already being displayed.
```

Reply:
Got it. I feel like you'll be doing a lot of same comments for the other two cuz we get those directly from the examples lol.